entrypoint.sh

```diff
@@ -5,20 +5,16 @@ set -e
 rm -f ./tmp/pids/server.pid
 mkdir -pm 1777 ./tmp
 
+# This is set to the name of the application for logging purposes
+PUBLIC_NAME="LR Landing"
+
 # Set the environment
 if [ -z "$RAILS_ENV" ]
 then
   export RAILS_ENV=production
 fi
 
-[ -z "$API_SERVICE_URL" ] && echo '{"ts": "'"$(date -u +%FT%T.%3NZ)"'", "level": "ERROR", "message": "You have not specified the env var API_SERVICE_URL"}' >&2
-
-[ -z "$APPLICATION_ROOT" ] && echo '{"ts": "'"$(date -u +%FT%T.%3NZ)"'", "level": "ERROR", "message": "You have not specified the env var APPLICATION_ROOT"}' >&2
-
-if [ -z "$API_SERVICE_URL" ] || [ -z "$APPLICATION_ROOT" ]
-then
-  exit 1
-fi
+echo "{\"ts\": $(date -u +%FT%T.%3NZ), \"level\": \"INFO\", \"message\": \"Initiating ${PUBLIC_NAME} application using APPLICATION_ROOT=${APPLICATION_ROOT}, API_SERVICE_URL=${API_SERVICE_URL}}"
```

Comment:
`APPLICATION_ROOT` is stale here.
API_SERVICE is redundant here.

Reply:
This echo was created as a verification flag for the build process and, while I agree and will remove the `APPLICATION_ROOT` variable, having the `API_SERVICE_URL` is good to know at this point in case you're needing to validate any local configuration being passed in at invocation.

Reply:
@andrew-pickin-epi Upon further reflection I've not removed the `APPLICATION_ROOT` from the application as it does allow us to pass in a temporary value, like `/`, when trying to get things running locally. 

I would like to say I think this makes the app more robust when developing locally even as an image as that will also match the [development.rb](https://github.com/epimorphics/lr-landing/blob/45608eea14eb2d60b4491f70715000dde287c511/config/environments/development.rb#L49) configuration settings:

```sh
  # The RAILS_RELATIVE_URL_ROOT should be set in the entrypoint.sh script via the 
  # build-time environment variable `APPLICATION_ROOT`; however, if this is not set
  # the following fallback is passed in as a standard root value when the app is run
  # directly via `rails server`
  config.relative_url_root = ENV.fetch('RAILS_RELATIVE_URL_ROOT', '/')
```

Therefore the echo above will allow us to verify the correct value is being used.

N.B. Please forgive me if I've gotten my `build-time` v. `run-time` variables mixed up nonetheless! 🙏

Reply:
I doesn't make it more robust.
If it's possible to set the variable differnetly at build time and run time the it can break,
IF you want to have two setups to run with the without a path then define different environment files. 
Further this IS a function of rails or at least rake. So No APPLICAITON_ROOT makes no sense at all.

Reply:
@andrew-pickin-epi while trying to resolve the proxy configuration mentioned previously via further investigation of the older versions of this application, and therefore also the other LR app configurations, we seem to have gotten things slightly muddled up in our approach. 

While I appreciate your guidance throughout these updates, and a lot of good choices were made, I would like to revert the `RAILS_RELATIVE_URL_ROOT` run-time variables to it's [previous configuration](https://github.com/epimorphics/lr-landing/blob/20cac1c7e4f16eeaa27186461faac26081b5bfaf/config/environments/production.rb#L88) based on my findings; let me explain.

Back in [January](https://github.com/epimorphics/lr-landing/commit/20cac1c7e4f16eeaa27186461faac26081b5bfaf) we discussed and agreed that the naming of the variables being passed around during the build process, specifically `RAILS_RELATIVE_URL_ROOT`, was misleading in its use in the build files. We decided then to rename that specific variable to ~~`APPLICATION_PATH`~~ `APPLICATION_ROOT` to be more clear on its purpose.

I, in my Rails naivety, misunderstood the use for that specific name and changed all instances of the RAILS/Rake variable used across the application as well. 

@der kindly highlighted my error to which I revised the renamed variable to only be the ones used in the build files, to which you then updated the validation methods you felt were appropriate in the `entrypoint.sh` file.

Upon further investigation, and in order to improve the quality of the codebase for the new infrastructure, we revised those validation flags together whilst also realising that the run-time value would work if the variable was "locked" to the expected `/app/root` path value in the rails production configuration file. We then cleaned up those validations to only check for the `API_SERVICE_URL` where applicable (funny enough, not in this application).

The Makefile configuration was also refactored as we realised the value being passed in during the precompile phase was being overwritten by the run-time locked value and therefore unnecessary.

While investigating the original configuration of the build files [[52](https://github.com/epimorphics/hmlr-linked-data/issues/52), [72](https://github.com/epimorphics/lr-landing/pull/72/files), [79](https://github.com/epimorphics/hmlr-linked-data/issues/79)] to sort out the proxy I've come to the conclusion that `APPLICATION_ROOT`, and subsequently `RAILS_RELATIVE_URL_ROOT`, should be available to the build process on ALL environments because if you're trying to run the docker image locally without having the proxy running you cannot see the styles, scripting, etc. if you can't change the `RAILS_RELATIVE_URL_ROOT` value; and if you try to run the rails server outside of docker in production mode to use the locked value, it won't build because the `RAILS_SECRET` isn't available.

Could we please discuss this together to agree on an approach that will allow for the most application resilience while still allowing for all of the environments to be run, tested, and validated both inside and out of docker images?

PS. I'm not averse to introducing specific environment files as suggested if that's the best way forward?